Assembler.Base/FramesAssembler.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using Assembler.Core;
+using Assembler.Core.Entities;
+using Assembler.Core.Enums;
+
+namespace Assembler.Base
+{
+    public class FramesAssembler<TFrame> : IAssembler<TFrame>
+        where TFrame : BaseFrame
+    {
+        private readonly IResolver<AssemblingPosition, IFrameHandler<TFrame>> _typeToHandlerResolver;
+        private readonly ILogger _logger;
+
+        public FramesAssembler(IResolver<AssemblingPosition, IFrameHandler<TFrame>> typeToHandlerResolver,
+            ILoggerFactory loggerFactory)
+        {
+            _typeToHandlerResolver = typeToHandlerResolver;
+            _logger = loggerFactory.GetLogger(this);
+        }
+
+        public void Assemble(TFrame frame)
+        {
+            try
+            {
+                IFrameHandler<TFrame> handler = _typeToHandlerResolver.Resolve(frame.AssemblingPosition);
+
+                _logger.Debug($"Frame [{frame.Guid}] is being sent to the handler.");
```

Comment:
I think you should move this and the next line out of the try clause (as a best practice when using try-catch)

Reply:
I don't know, it makes the code really messy.
![image](https://user-images.githubusercontent.com/3995473/78040803-9c134600-7378-11ea-8107-e7f39a56d79c.png)

Reply:
private methods

Reply:
f

Reply:
@IdanZel , if you think thats fine, you can resolve. I agree with the solution.